### PR TITLE
building a sandbox image using geobase builder and runner

### PR DESCRIPTION
**Pull request by NikitaGandhi**
- use geobase-builder and geobase-runner to build new image
- added constraints and nobinary files for each requirements.txt

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -43,57 +44,59 @@ RUN echo "Enable server extensions" \
...
-  libboost-system1.65.1 \
-  libboost-filesystem1.65.1 \
-  libboost-date-time1.65.1 \
+# TODO -  Unable to locate package?
+#  libcgal13 \
+#  libmuparser2v5 \
+#  libgsl23 \
+#  libboost-system1.65.1 \
+#  libboost-filesystem1.65.1 \
+#  libboost-date-time1.65.1 \
```
Comment (NikitaGandhi) [inline docker/Dockerfile]: Not able to install these packages - @Kirill888 @alexgleith can you help?

File: docker/requirements.txt [hunk 2]
```diff
@@ -1,116 +1,100 @@
...
-,msgpack\
-,multidict\
-,statsmodels\
-,typed_ast\
-,tables\
-,zstandard\
-,richdem\
+https://github.com/ubarsc/rios/archive/rios-1.4.10.tar.gz
+# TODO - ?
+# https://dl.google.com/coral/python/tflite_runtime-2.1.0.post1-cp36-cp36m-linux_x86_64.whl
```
Comment (NikitaGandhi) [inline docker/requirements.txt]: this also didn't work

Comment (NikitaGandhi) [reply docker/requirements.txt:100]: installed libcgal-dev and libboost-dev instead

File: docker/Dockerfile (hunk 1 above)
Comment (NikitaGandhi) [reply docker/Dockerfile]: installed libcgal-dev and libboost-dev instead

File: docker/requirements.txt (hunk 2 above)
Comment (Kirill888) [reply docker/requirements.txt]: this is also expected, new image has newer python so you'll need to use `-cp38` version of tflite

**Comment by Kirill888** [comment]
@NikitaGandhi new builder/runner are based on a newer Ubuntu image so I would expect things like that to fail.

Honestly I think the current approach is kinda broken. `geobase` images are useful for deployment of " focused tools", that only need datacube and maybe odc-tools in them. For a complex system like sandbox the size advantage you get by using builder/runner split is negligible. I would recommend switching to a single stage docker basing it on the builder image, or even moving to conda and repo2docker type of solution.

Also have a look here:

https://github.com/opendatacube/geobase/tree/develop/odc/odc-dev

**Comment by Kirill888** [comment]
@omad how about building one single conda environment deployable to both NCI and sandbox, just need to use nci compatible paths when baking that into sandbox docker.

@NikitaGandhi also I believe Robert Woodcock might have python 3.8 based sandbox somewhere.

File: docker/requirements.txt (hunk 2 above)
Comment (Kirill888) [reply docker/requirements.txt]: same story with `rsgislib` I compiled it ages ago for py36 only, it won't work with py38

Comment (NikitaGandhi) [reply docker/requirements.txt]: did this
```
--extra-index-url="https://google-coral.github.io/py-repo/"
tflite_runtime
```

**Comment by NikitaGandhi** [comment]
> @omad how about building one single conda environment deployable to both NCI and sandbox, just need to use nci compatible paths when baking that into sandbox docker.
> 
> @NikitaGandhi also I believe Robert Woodcock might have python 3.8 based sandbox somewhere.

I have created new issue ticket for discussion and consideration - https://github.com/GeoscienceAustralia/dea-sandbox/issues/183

**Comment by NikitaGandhi** [comment]
@alexgleith this PR is ready to merge. Tested with deafrica-dev sandbox environment

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,26 +1,26 @@
-ARG V_BASE=-3.0.4
+ARG V_BASE=3.3.0
 ARG py_env_path=/env
```
Comment (alexgleith) [inline docker/Dockerfile]: I think we should _only_ have this below the `FROM` command

**Review by alexgleith — APPROVED** [review]
I think you've done an amazing job getting this over the line, @NikitaGandhi !

**Review by mmochan — APPROVED** [review]
Great job - LGTM!

**Pull request merged**